### PR TITLE
Return warnings to log when NuGet SDK resolver fails

### LINKED ISSUE #12049: [Bug]: NugetSDKResolver doesn't give detailed error messages when it fails
### NuGet Product Used

MSBuild.exe, NuGet.exe, NuGet SDK

### Product Version

All

### Worked before?

No

### Impact

It bothers me. A fix would be nice

### Repro Steps & Context

Try to use the NugetSdkResolver but with a package it can't load. It will fail but not tell you why, making it hard to diagnose the problem.

### Verbose Logs

_No response_

### PR DESCRIPTION
**Pull request by Forgind**
The NuGet SDK resolver should return warnings for MSBuild to log when it fails to resolve an SDK. This helps users properly diagnose what went wrong.

## Bug

Fixes: https://github.com/NuGet/Home/issues/12049

Regression? Last working version:

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [ ] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

**Comment by nkolev92** [comment]
Hey @Forgind 
Can you please create a matchign issue and fill out the template?

File: src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs [hunk 1]
```diff
@@ -73,14 +73,14 @@ public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext
...
-                return null;
+                return factory.IndicateFailure(null, new List<string>() { "NuGetSdkResolver did not resolve this SDK because it was disabled by the MSBUILDDISABLENUGETSDKRESOLVER environment variable." });
             }
 
             // This resolver only works if the user specifies a version in a project or a global.json.
             // Ignore invalid versions, there may be another resolver that can handle the version specified
             if (!TryGetNuGetVersionForSdk(sdkReference.Name, sdkReference.Version, resolverContext, out var parsedSdkVersion))
             {
-                return null;
+                return factory.IndicateFailure(null, new List<string>() { "NuGetSdkResolver did not resolve this SDK because there was no version specified in the project or global.json." });
```
Comment (nkolev92) [inline src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs]: We probably want to localize this no?

**Review by nkolev92 — COMMENTED** [review]
I think loc concerns apply to both. 

To localize, add a resource to Strings in the project.

File: src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs [hunk 2]
```diff
@@ -73,14 +73,14 @@ public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext
             // Escape hatch to disable this resolver
             if (DisableNuGetSdkResolver.Value)
             {
-                return null;
+                return factory.IndicateFailure(null, new List<string>() { "NuGetSdkResolver did not resolve this SDK because it was disabled by the MSBUILDDISABLENUGETSDKRESOLVER environment variable." });
```
Comment (dominoFire) [inline src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs]: Per coding guidelines, use named arguments on null arguments to clarify context.

```suggestion
                return factory.IndicateFailure(errors: null, warnings: new List<string>() { Strings.Error_DisabledSdkResolver });
```

Later, modify [Strings.resx](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/Strings.resx) file to include `Error_DisabledSdkResolver` string resource

File: src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs (hunk 1 above)
Comment (dominoFire) [inline src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs]: Per coding guidelines, use named arguments on null arguments to clarify context.

```suggestion
                return factory.IndicateFailure(errors: null, warnings: new List<string>() { Strings.Error_NoSdkVersion });
```

Later, modify [Strings.resx](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/Strings.resx) file to include `Error_NoSdkVersion` string resource

**Review by dominoFire — COMMENTED** [review]
Added some comments. Thank you.

**Review by nkolev92 — APPROVED** [review]

**Review by jeffkl — APPROVED** [review]

**Review by zivkan — APPROVED** [review]

**Pull request merged**